arch/csr/dcsr.yaml

```
@@ -127,34 +127,70 @@ fields:
     reset_value: 0
   STEPIE:
     location: 11
-    type: RW
     description: |
       0 (interrupts disabled):: Interrupts (including NMI) are disabled during single stepping with step set.
       This value should be supported.
       1 (interrupts enabled):: Interrupts (including NMI) are enabled during single stepping with step set.
       Implementations may hard wire this bit to 0. In that case interrupt behavior can be emulated by the
       debugger. The debugger must not change the value of this bit while the hart is running.
-    reset_value: 0
+    type(): |
+      if (DCSR_STEPIE_TYPE == "tied-to-0" || DCSR_STEPIE_TYPE == "tied-to-1") {
+        return CsrFieldType::RO;
+      } else if (DCSR_STEPIE_TYPE == "rw") {
+        return CsrFieldType::RW;
+      }
```

```suggestion
      return CsrFieldType::RW;
```
Personal preference, but I like to see that functions always return _something_. I know we have all the cases covered, but that's not obvious from just looking at this code...you'd have to go look at the definition of `DSCR_STEPIE_TYPE`.